include/deal.II/matrix_free/operators.h

```diff
@@ -35,6 +35,345 @@ DEAL_II_NAMESPACE_OPEN
 
 namespace MatrixFreeOperators
 {
+  // first some auxiliary functions we need to handle
+  // VectorTypes and BlockVectorTypes simultaneously
+  namespace
+  {
+    template <int dim, typename VectorType>
+    typename std_cxx11::enable_if<IsBlockVector<VectorType>::value, void>::type
+    external_initialize_dof_vector(const MatrixFree<dim, typename VectorType::value_type> &mf,
+                                   VectorType &vec)
+    {
+      for (unsigned int i = 0; i < vec.n_blocks(); ++i)
+        {
+          if (!vec.block(i).partitioners_are_compatible(*mf.get_dof_info(i).vector_partitioner))
```

Comment:
looking again at all these auxiliary functions, i think we can make them go by
```
for (unsigned int i = 0; i < internal_n_blocks(vec); ++i)
   {
        internal_get_block(vec,i).whatever...
   }
```
The difference appears to be only in accessing the vector directly or it's block.